### PR TITLE
extension toLowerCase

### PR DESCRIPTION
**Pull request by forresto**
#1

### CODE REVIEW

**Comment by forresto** [comment]
spec to come... just figured out the environment variable idea

**Comment by forresto** [comment]
Or not... I can't see them in Travis. Seems like we could use "key" and "secret" since we're just testing url generation, not that they actually work.

File: spec/imgflo-url.coffee [hunk 1]
```diff
@@ -152,6 +156,19 @@ describe 'imgflo-url', ->
 
           expect(url).to.equal "https://imgflo.herokuapp.com/graph/#{process.env.IMGFLO_KEY}/#{process.env.IMGFLO_TOKEN_03}/passthrough.jpg?input=https%3A%2F%2Fv.cdn.vine.co%2Fr%2Fvideos%2FB5B06468B91176403722801139712_342c9a1c624.1.5.15775156368984795444.mp4.jpg%3FversionId%3DedU_LrAtIFsGvZj.Fgi0Si1bem68tBlk"
 
+      context 'with a url with uppercase JPG extension', ->
+
+        it 'should produce the correct URL', ->
+          config = getConfig()
```
Comment (forresto) [inline spec/imgflo-url.coffee:144]: @paulyoung I can make the other specs match this with the dumb key, secret.

Or you can set up `process.env.IMGFLO_TOKEN_04`.

Comment (paulyoung) [inline spec/imgflo-url.coffee:144]: Is there any concern with exposing tokens like that?

If we can remove the environment variables altogether, that would be great.

File: spec/imgflo-url.coffee [hunk 2]
```diff
@@ -152,6 +156,33 @@ describe 'imgflo-url', ->
...
+
+          params =
+            input: 'http://1.bp.blogspot.com/-1h8jX1lfGJc/UnlTZ3Fsq6I/AAAAAAAAAOo/dw7IXnJBO5A/s1600/IMG_1722.JPG'
+
+          url = imgflo config, 'passthrough', params
+
+          expect(url).to.contain 'passthrough.jpg'
```
Comment (paulyoung) [inline spec/imgflo-url.coffee:151]: Is this redundant or just to fail early?

File: index.coffee [hunk 3]
```diff
@@ -29,7 +31,9 @@ imgflo = (config, graph, params, format) ->
   throw new Error 'imgflo params must contain an "input" key' unless input?
 
   parsed = url.parse input
-  extension = path.extname(parsed.pathname).match(/^\.(\w+)/)[1]
+  match = path.extname(parsed.pathname).match(/^\.(\w+)/)
+  if match?
+    extension = match[1].toLowerCase()
```
Comment (paulyoung) [inline index.coffee]: Could we change this to `extension = match?[1].toLowerCase()` instead?

File: spec/imgflo-url.coffee (hunk 2 above)
Comment (forresto) [inline spec/imgflo-url.coffee:151]: Redundant for readability.

File: spec/imgflo-url.coffee (hunk 1 above)
Comment (forresto) [inline spec/imgflo-url.coffee:144]: That token is generated from "key" and "secret" strings, so it won't resolve to an image, but it does test the right thing.

File: index.coffee (hunk 3 above)
Comment (forresto) [inline index.coffee]: That seems to be more the style 'round here.

**Comment by paulyoung** [comment]
@forresto let me know if you push more commits.

**Comment by forresto** [comment]
:boom:

**Comment by paulyoung** [comment]
Only thing I would change is to use `getConfig` everywhere in the specs but that's a bit of a nitpick.

**Comment by paulyoung** [comment]
Thanks @forresto!

**Pull request merged**